### ISSUE TITLE
@UseKafkaClient is not working with TestSuite

### ISSUE THREAD

**Issue by PrabaharanK** [authorjapps/zerocode#317]
I'm using zerocode-tdd version of 1.3.8. When I write individual test classes with customized kafka client  like below 

```java
@UseKafkaClient(CustomKafkaClient.class)
public class TestClass {
    @Test
    @JsonTestCase("scenarioJsons/json1.json")
    public void testMethod1() throws Exception {
    }

 }
```

My CustomKafkaClient.class is loaded and everything works fine. 

When I try to run the Json tests through TestSuite like below 
```java
@TargetEnv("properties/server.properties")
@TestPackageRoot("scenarioJsons")
@EnvProperty("_${env}")
@RunWith(ZeroCodePackageRunner.class)
@UseKafkaClient(CustomKafkaClient.class)
public class SmokeTestSuite {}
```

Though I annotated test suite class with @UseKafkaClient(CustomKafkaClient.class), BasicKafkaClient.class from org.jsmart.zerocode.core.kafka.client is loaded instead of CustomKafkaClient.class due to that my Kafka consumers are failing.

Can anyone help me to load the customized CustomKafkaClient.class in test suite scenario ?

**Comment by authorjapps** [comment]
@PrabaharanK , yes, we can help you to sort out this.
Can you please join our Mailing-List and Slack channel and ping us there... 
You might get a better/faster response there.

**Comment by officiallysameer** [comment]
@PrabaharanK 
How urgent is this issue is for you? Can you provide your use-case in more detail so that we can try to help you better

**Comment by PrabaharanK** [comment]
This s not urgent to me. use case is I'm preparing TestSuite,I want to add customized kafka client to send and receive messages from topic. Problem is, with @RunWith(ZeroCodePackageRunner.class)  my customized kafka client is not working. Test suite took default BasicKafkaClient.class. 

Scenario, 
1. Create Test Suite like below
@TargetEnv("properties/server.properties") @TestPackageRoot("scenarioJsons") @EnvProperty("_${env}") @RunWith(ZeroCodePackageRunner.class) @UseKafkaClient(CustomKafkaClient.class) public class SmokeTestSuite {}

2. add Customized kafka client by @UseKafkaClient(CustomKafkaClient.class)
3. if start your test suite BasicKafkaClient.class instead of picking my Customized one.

**Comment by authorjapps** [comment]
> I want to add customized kafka client to send and receive messages from topic....

Hello @PrabaharanK , thanks. The point of the question here is - we are trying to understand why the `custom client` is needed for your project. Probably we can implement the features into the `BasicKafkaClient.class` if that helps.

**Comment by PrabaharanK** [comment]
The Point is, Messages we pushed and received from Kafka topic is encrypted. The Encryption  and decryption mechanisms are company specific. So Whatever message we push should be encrypted and vice versa.

**Comment by nirmalchandra** [comment]
Thanks @PrabaharanK , makes sense now.
I am working on this and will be raising the PR soon with the fix.

**Comment by authorjapps** [comment]
Available in version `1.3.16 `

**Comment by authorjapps** [comment]
Hello @PrabaharanK mate, would you please join the Slack channel n the mailing list? 

Link: [Join Slack](https://join.slack.com/t/zerocode-workspace/shared_invite/enQtNzYxMDAwNTQ3MjY1LTA2YmJjODJhNzQ4ZjBiYTQwZDBmZmNkNmExYjA3ZDk2OGFiZWFmNWJlNGRkOTdiMDQ4ZmQyNzcyNzVjNWQ4ODQ)

Link: [Join Mailing list](https://groups.google.com/forum/#!forum/zerocode-automation)
